### PR TITLE
Updates to mongo getter to reduce duplicate code

### PR DESCRIPTION
**Pull request by bou1der**
**What is the problem I am trying to address?**

In the mongo/getter implementation there is duplicated code in the Info and GoMod functions. The purpose of this PR is to reduce the duplicated code without any changes to the operation of either of the functions. 

There was also a difference between how the Info and GoMod functions passed the reference for the query result to the Decode function. The Info passed a pointer to the reference while the GoMod function just passed the reference. Mongo handles both so there was no error during run time but this refactor ensures that Info and GoMod call the mongo.Decode function the same way.

**How is the fix applied?**

The common/duplicated code from Info and GoMod functions has been moved into a new getter.Query function.

**Mention the issue number it fixes or add the details of the changes if it doesn't have a specific issue.**

Fixes # (No Existing Issue)

### CODE REVIEW

File: pkg/storage/mongo/getter.go [hunk 1]
```diff
@@ -16,26 +16,11 @@ import (
 // Info implements storage.Getter
 func (s *ModuleStore) Info(ctx context.Context, module, vsn string) ([]byte, error) {
 	const op errors.Op = "mongo.Info"
-	ctx, span := observ.StartSpan(ctx, op.String())
```
Comment (manugupt1) [inline pkg/storage/mongo/getter.go]: Can you keep this span here to collect traces?

File: pkg/storage/mongo/getter.go [hunk 2]
```diff
@@ -44,24 +29,11 @@ func (s *ModuleStore) Info(ctx context.Context, module, vsn string) ([]byte, err
 // GoMod implements storage.Getter
 func (s *ModuleStore) GoMod(ctx context.Context, module, vsn string) ([]byte, error) {
 	const op errors.Op = "mongo.GoMod"
-	ctx, span := observ.StartSpan(ctx, op.String())
```
Comment (manugupt1) [inline pkg/storage/mongo/getter.go]: Please leave it as it is to collect traces?

File: pkg/storage/mongo/getter.go [hunk 3]
```diff
@@ -91,3 +63,30 @@ func (s *ModuleStore) Zip(ctx context.Context, module, vsn string) (io.ReadClose
 
 	return dStream, nil
 }
+
+// Query connects and queries storage module
+func Query(ctx context.Context, s *ModuleStore, module, vsn string, op errors.Op) (*storage.Module, error) {
```
Comment (manugupt1) [inline pkg/storage/mongo/getter.go]: Can you have it as `func (s* moduleStore) query(ctx, module, .....)`  this way query will be private to the package?

Comment (bou1der) [reply pkg/storage/mongo/getter.go]: Definitely, that makes a lot more sense.

File: pkg/storage/mongo/getter.go (hunk 2 above)
Comment (bou1der) [reply pkg/storage/mongo/getter.go]: No problem. Thanks for taking the time to review these changes!

File: pkg/storage/mongo/getter.go (hunk 1 above)
Comment (bou1der) [reply pkg/storage/mongo/getter.go]: No problem - Should I also start a trace for the new function (query) or will the trace on Info capture the desired metrics?

Comment (manugupt1) [reply pkg/storage/mongo/getter.go]: yes! that is super important imo as that is where we will do majority of compute time. By having traces here, we can see which function the query was a subset of

```
----Info----- ------GoMod------
 --query---     ----query---
```

**Comment by arschles** [comment]
@manugupt1 looks like @bou1der made your requested changes, so I'm going to click the "dismiss review" button so we can get this merged

**Pull request merged**